### PR TITLE
[Enhancement]grouping function support expression

### PR DESCRIPTION
**Pull request by even986025158**
## Why I'm doing:
https://github.com/StarRocks/starrocks/issues/50136
## What I'm doing:
add cast expression into analyzeState
Fixes #issue
#50136
## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java [hunk 1]
```diff
@@ -1296,7 +1296,7 @@ public Void visitGroupingFunctionCall(GroupingFunctionCallExpr node, Scope scope
             if (node.getChildren().size() < 1) {
                 throw new SemanticException("GROUPING functions required at least one parameters", node.getPos());
             }
-            if (node.getChildren().stream().anyMatch(e -> !(e instanceof SlotRef))) {
+            if (node.getChildren().stream().anyMatch(e -> !(e instanceof SlotRef || e instanceof CastExpr))) {
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java:1299]: don't need check only cast

File: fe/fe-core/src/test/java/com/starrocks/connector/analyzer/SimpleExpressionAnalyzer.java [hunk 2]
```diff
@@ -660,8 +660,8 @@ public Void visitGroupingFunctionCall(GroupingFunctionCallExpr node, Scope scope
             if (node.getChildren().size() < 1) {
                 throw new SemanticException("GROUPING functions required at least one parameters");
             }
-            if (node.getChildren().stream().anyMatch(e -> !(e instanceof SlotRef))) {
-                throw new SemanticException("grouping functions only support column.");
+            if (node.getChildren().stream().anyMatch(e -> !(e instanceof SlotRef || e instanceof CastExpr))) {
```
Comment (Seaven) [inline fe/fe-core/src/test/java/com/starrocks/connector/analyzer/SimpleExpressionAnalyzer.java:663]: don't need check cast

**Comment by Seaven** [comment]
add more test case

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java (hunk 1 above)
Comment (even986025158) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java]: it has updated

**Review by Seaven — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Pull request merged**